File: 41. First Missing Positive/sol.py
class Solution(object):
    def firstMissingPositive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = {}
        for i in range(1,len(nums)+2):
            res[i] = 0
        for i in nums:
            if i > 0 and i <= len(nums):
                res[i] = 1
        
        for i in res:
            if res[i] == 0:
                return i

File: 50. Pow(x, n)/sol.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double res = 1;
        int p = abs(n);
        while(p > 0){
            if(p % 2 == 1){
                res = (res * x);
            }
            x = x * x;
            p = p / 2;
        }
        if (n < 0){
            return 1.0/res;
        }
        return res;
    }
};